### ISSUE TITLE
Error while installing lxml

### ISSUE THREAD

**Issue by Avi30000** [odoo/odoo#163370]
Please someone help to fix this for me. 

C:\Python310\python.exe C:\Users\Sequel\Downloads\odoo-16.0\odoo-16.0\odoo-bin -c odoo.conf 
Traceback (most recent call last):
  File "C:\Users\Sequel\Downloads\odoo-16.0\odoo-16.0\odoo-bin", line 5, in <module>
    import odoo
  File "C:\Users\Sequel\Downloads\odoo-16.0\odoo-16.0\odoo\__init__.py", line 134, in <module>
    from . import modules
  File "C:\Users\Sequel\Downloads\odoo-16.0\odoo-16.0\odoo\modules\__init__.py", line 8, in <module>
    from . import db, graph, loading, migration, module, registry, neutralize
  File "C:\Users\Sequel\Downloads\odoo-16.0\odoo-16.0\odoo\modules\graph.py", line 10, in <module>
    import odoo.tools as tools
  File "C:\Users\Sequel\Downloads\odoo-16.0\odoo-16.0\odoo\tools\__init__.py", line 24, in <module>
    from .mail import *
  File "C:\Users\Sequel\Downloads\odoo-16.0\odoo-16.0\odoo\tools\mail.py", line 32, in <module>
    safe_attrs = clean.defs.safe_attrs | frozenset(
AttributeError: module 'lxml.html.clean' has no attribute 'defs'

Process finished with exit code 1

**Comment by lathama** [comment]
The lxml package was broken up recently install lxml-html-clean to fix the issue.

* https://github.com/odoo/docker/issues/498
* https://pypi.org/project/lxml-html-clean/

**Comment by lathama** [comment]
Related https://github.com/odoo/odoo/issues/163429

**Comment by lathama** [comment]
Related https://github.com/odoo/odoo/issues/157950

**Comment by lathama** [comment]
TL;DR;
Install https://tracker.debian.org/pkg/lxml-html-clean or https://pypi.org/project/lxml-html-clean/

**Comment by Avi30000** [comment]
Thanks a lot, appreciate it, this is the first time am trying without any knowledge or experience of python, pycharm, etc.

**Comment by vava-odoo** [comment]
Thanks @lathama

**Comment by leio** [comment]
Installing `python3-lxml-html-clean` on e.g. Ubuntu 24.04 LTS is not sufficient, as with it `lxml.html.clean.defs` does not exist as it's in a separate module now. Odoo should have accessed `html.defs` instead of `clean.defs` in the first place in `odoo/tools/mail.py`, which should work with old `lxml` as well. I believe `clean.defs` working with old lxml is more of an accident, as e.g. help(clean.defs) after doing the same imports points at `lxml.html.defs` documentation.

No PR will be forthcoming from me, as I will not be signing a CLA, but hopefully it helps someone stumbling on this.

**Comment by leio** [comment]
I found #161366 existing for this already.

**Comment by astrojuanlu** [comment]
Shouldn't this be included in `setup.py` then?
